src/hotspot/share/gc/serial/serialHeap.inline.hpp

```diff
@@ -31,12 +31,20 @@
 #include "gc/serial/tenuredGeneration.inline.hpp"
 
 template <typename OopClosureType1, typename OopClosureType2>
-void SerialHeap::oop_since_save_marks_iterate(OopClosureType1* cur,
-                                              OopClosureType2* older) {
+inline void SerialHeap::oop_since_save_marks_iterate(OopClosureType1* cur,
+                                                     OopClosureType2* older) {
   young_gen()->oop_since_save_marks_iterate(cur);
   old_gen()->oop_since_save_marks_iterate(older);
 }
 
+inline void SerialHeap::scan_evacuated_objs(YoungGenScanClosure* young_cl,
+                                            OldGenScanClosure* old_cl) {
+  do {
+    oop_since_save_marks_iterate(young_cl, old_cl);
```

Comment:
I meant physically copy the body of `SerialHeap::oop_since_save_marks_iterate` here.

Reply:
> I meant physically copy the body of SerialHeap::oop_since_save_marks_iterate here.

OK.

> I inlined both oop_since_save_marks_iterate and scan_evacuated_objs. But it seems that you only want to inline oop_since_save_marks_iterate?

What do you think about that I marked `scan_evacuated_objs` as `inline`?

Reply:
No need for `scan_evacuated_objs` to be `inline`, I believe. The expensive part is the loop.